### PR TITLE
Expose some things for ns2.0

### PR DESCRIPTION
**Pull request by kouvel**
Mostly:
- Exposed flow-related methods on `ExecutionContext`
- Enabled and exposed `SynchronizationContext` wait overriding

Realted issues:
- dotnet/corefx#11638
- dotnet/corefx#11633
- dotnet/corefx#11635
- dotnet/corefx#11636
- dotnet/corefx#13484

### CODE REVIEW

**Comment by kouvel** [comment]
@rahku @gkhanna79 @jkotas

File: src/mscorlib/src/System/Threading/ExecutionContext.cs [hunk 1]
```diff
@@ -315,6 +380,72 @@ internal bool IsPreAllocatedDefault
...
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is AsyncFlowControl && Equals((AsyncFlowControl)obj);
+        }
+
+        public bool Equals(AsyncFlowControl obj)
+        {
+            return true;
```
Comment (rahku) [inline src/mscorlib/src/System/Threading/ExecutionContext.cs]: why is this always true

File: clr.coreclr.props [hunk 2]
```diff
@@ -51,6 +51,7 @@
     <FeatureUseAsmGCWriteBarriers>true</FeatureUseAsmGCWriteBarriers>
     <!-- Setting this to "false" works only for workstation GC, not server. -->
     <FeatureSymDiff>true</FeatureSymDiff>
+    <FeatureSynchronizationContextWait>true</FeatureSynchronizationContextWait>
```
Comment (rahku) [inline clr.coreclr.props]: changing this file may not be required anymore.

File: src/mscorlib/src/System/Threading/ExecutionContext.cs (hunk 1 above)
Comment (kouvel) [inline src/mscorlib/src/System/Threading/ExecutionContext.cs]: Oops, I had implemented a no-op version first, need to fix

**Comment by kouvel** [comment]
@dotnet-bot retest Linux ARM Emulator Cross Release Build

**Comment by kouvel** [comment]
Does this look ok to merge after tests pass?

**Comment by rahku** [comment]
LGTM

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Pull request merged**